SignalServiceKit/src/Messages/Attachments/TSAttachmentStream.m

```diff
@@ -610,6 +613,46 @@ - (CGFloat)audioDurationSeconds
     return audioDurationSeconds;
 }
 
+- (nullable OWSBackupFragment *)lazyRestoreFragment
+{
+    if (!self.lazyRestoreFragmentId) {
+        return nil;
+    }
+    return [OWSBackupFragment fetchObjectWithUniqueID:self.lazyRestoreFragmentId];
+}
+
+#pragma mark - Update With... Methods
+
+- (void)markForLazyRestoreWithFragment:(OWSBackupFragment *)lazyRestoreFragment
+                           transaction:(YapDatabaseReadWriteTransaction *)transaction
+{
+    OWSAssert(lazyRestoreFragment);
+    OWSAssert(transaction);
+
+    if (!lazyRestoreFragment.uniqueId) {
+        // If metadata hasn't been saved yet, save now.
+        [lazyRestoreFragment saveWithTransaction:transaction];
+
+        OWSAssert(lazyRestoreFragment.uniqueId);
+    }
+    [self applyChangeToSelfAndLatestCopy:transaction
+                             changeBlock:^(TSAttachmentStream *attachment) {
+                                 [attachment setLazyRestoreFragmentId:lazyRestoreFragment.uniqueId];
+                             }];
+}
+
+- (void)updateWithLazyRestoreComplete
+{
+    [self.dbReadWriteConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *transaction) {
+        [self applyChangeToSelfAndLatestCopy:transaction
+                                 changeBlock:^(TSAttachmentStream *attachment) {
+                                     [attachment setLazyRestoreFragmentId:nil];
+                                 }];
+    }];
+
+    [self ensureThumbnail];
```

Comment:
Is this necessary? 

Won't this happen by virtue of `applyChangeToSelf...` calling  `saveWithTransaction:`?

Reply:
Oh great!